sql/src/test/java/io/crate/metadata/doc/DocIndexMetaDataTest.java

```diff
@@ -1002,6 +1007,91 @@ public void testNewArrayMapping() throws Exception {
                 is((DataType) new ArrayType(DataTypes.TIMESTAMP)));
     }
 
+    @Test
+    public void testPartitionInfosFreshTable() throws Exception {
+        DocIndexMetaData md = getDocIndexMetaDataFromStatement("create table foo (id long, type byte) partitioned by (type)");
+        assertThat(md.partitionInfos(), is(notNullValue()));
+        assertThat(md.partitionInfos().size(), is(0));
+    }
+
+    @Test
+    public void testMergePartitionWithDifferentShardsAndReplicas() throws Exception {
+        XContentBuilder builder = XContentFactory.jsonBuilder()
```

Comment:
why such a complicated structure just to test the different shard and replica aspects
can you use sql for this?